src/test/java/com/solvd/ecommerce/EcommerceTest.java

```diff
@@ -1,40 +1,84 @@
 package com.solvd.ecommerce;
 
+import com.solvd.ecommerce.page.Header;
 import com.solvd.ecommerce.page.HomePage;
+import com.solvd.ecommerce.page.LoginPage;
 import com.solvd.ecommerce.page.ResultPage;
-import com.solvd.ecommerce.utils.ConfigReader;
-import io.github.bonigarcia.wdm.WebDriverManager;
-import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
 import org.testng.Assert;
-import org.testng.annotations.AfterTest;
-import org.testng.annotations.BeforeTest;
-import org.testng.annotations.Test;
+import org.testng.annotations.*;
+import org.testng.asserts.SoftAssert;
 
-public class EcommerceTest {
+import java.util.List;
 
-    private WebDriver webDriver;
+public class EcommerceTest extends AbstractTest {
 
-    @BeforeTest
-    public void setupWebDriver() {
-        WebDriverManager.chromedriver().setup();
+    @Test(dataProvider = "query")
+    public void verifySearchResultTest(String query) {
+        ResultPage resultPage = new ResultPage(getDriver());
+        Header header = new Header(getDriver());
+        header.searchInputEnter(query);
+        header.searchButtonClick();
+        Assert.assertNotEquals(resultPage.getResultListSize(), 0);
+        resultPage.printItemData();
     }
 
     @Test
-    public void verifySearchResultTest() {
-        String query = "Maped";
-        webDriver = new ChromeDriver();
-        webDriver.get(ConfigReader.getValue("URL"));
-        HomePage homePage = new HomePage(webDriver);
-        ResultPage resultPage = new ResultPage(webDriver);
-        homePage.searchInputEnter(query);
-        homePage.searchButtonClick();
-        Assert.assertNotEquals(resultPage.getResultListSize(), 0);
-        resultPage.printItemData();
+    public void checkBackgroundChangeTest() {
+        HomePage homePage = new HomePage(getDriver());
+        Actions action = new Actions(getDriver());
+        SoftAssert sa = new SoftAssert();
+        List<WebElement> list = homePage.getSidebarMenuList();
+        for (WebElement el : list) {
+            action.moveToElement(el).perform();
+            sa.assertTrue(el.getAttribute("class").contains("main-nav__list__li_wnav_active"), "List element hasn't changed appearance on hover.");
+        }
```

Comment:
All types of work with elements must be on pages on in the tests. 
You need to create methods on necessary page and in test just call necessary test methods